packages/zoe/src/contracts/oracle.js

```diff
@@ -0,0 +1,145 @@
+// @ts-check
+import { assert, details } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+import { trade } from '../contractSupport';
+
+import '../../exported';
+
+/**
+ * This contract provides oracle queries for a fee or for pay.
+ *
+ * @type {ContractStartFn}
+ *
+ */
+const start = async zcf => {
+  const { oracleDescription } = zcf.getTerms();
+
+  const {
+    brands: { Fee: feeBrand },
+    maths: { Fee: feeMath },
+  } = zcf.getTerms();
+
+  /** @type {OracleHandler} */
+  let handler;
+  /** @type {string} */
+  const description = oracleDescription;
+
+  const { zcfSeat: feeSeat } = zcf.makeEmptySeatKit();
+
+  let lastIssuerNonce = 0;
+  /** @type {string} */
+  let revoked;
+
+  /** @type {OracleCreatorFacet} */
+  const realCreatorFacet = {
+    async addFeeIssuer(issuerP) {
+      lastIssuerNonce += 1;
+      const keyword = `OracleFee${lastIssuerNonce}`;
+      await zcf.saveIssuer(issuerP, keyword);
+    },
+    makeWithdrawInvitation(total = false) {
+      return zcf.makeInvitation(seat => {
+        const gains = total
+          ? feeSeat.getCurrentAllocation()
+          : seat.getProposal().want;
+        trade(zcf, { seat: feeSeat, gains: {} }, { seat, gains });
+        seat.exit();
+        return 'liquidated';
+      }, 'oracle liquidation');
+    },
+    getCurrentFees() {
+      return feeSeat.getCurrentAllocation();
+    },
+  };
+
+  const creatorFacet = harden({
+    initialize(privateParams) {
+      const { oracleHandler } = privateParams;
+      handler = oracleHandler;
+      return realCreatorFacet;
+    },
+  });
+
+  /** @type {OraclePublicFacet} */
+  const publicFacet = harden({
+    getDescription() {
+      return description;
+    },
+    async query(query) {
+      try {
+        if (revoked) {
+          throw Error(revoked);
+        }
```

Comment:
If we do a `zcf.shutdown` when the oracle has been revoked, this call should no longer work, since the publicFacet is in a shutdown vat. Unfortunately, we cannot test this without using Swingset tests. We should open a general Zoe issue to allow for testing whether a vat has been shutdown without using a full Swingset test. 

Long story short, this `revoked` should be unnecessary if we do a vat.shutdown, but we may wish to keep it because the tests will not mock a shutdown vat appropriately.

We can also use assert:
```suggestion
        assert(!revoked, revoked)
```